frontend/src/components/Dropdown/components/MultiSelectMenu.tsx

```diff
@@ -40,14 +49,23 @@ export const MultiSelectMenu = (): JSX.Element => {
         })}
         sx={styles.list}
       >
-        {isOpen &&
-          items.map((item, index) => (
-            <MultiDropdownItem
-              key={`${itemToValue(item)}${index}`}
-              item={item}
-              index={index}
-            />
-          ))}
+        {isOpen && items.length > 0 && (
+          <Virtuoso
+            ref={virtualListRef}
+            data={items}
+            overscan={VIRTUAL_LIST_OVERSCAN_HEIGHT}
+            style={{ height: virtualListHeight }}
+            itemContent={(index, item) => {
+              return (
+                <MultiDropdownItem
+                  key={`${itemToValue(item)}${index}`}
+                  item={item}
+                  index={index}
+                />
+              )
+            }}
+          />
+        )}
         {isOpen && items.length === 0 ? (
```

Comment:
extreme nit: I might restructure this to be

```javascript
{ isOpen && (
  items.length === 0 
    ? <ListItem...> {nothingFoundLabel} </ListItem>
    : <Virtuoso ... /> 
)}

Reply:
i find these nested tertiary conditionals hard to read and parse haha. Conscious decision for readability.

Reply:
haha its fine, for me it helps the train of logic more when I can see i've covered all cases (in this case, the reason I prefer it is because if the dropdown is open, then i know i've covered both the ===0 and >0 case. and its obvious that if the dropdown is closed, we don't want to render anything)